pennylane/_device.py

```diff
@@ -322,6 +322,24 @@ def execute(self, queue, observables, parameters={}, **kwargs):
 
             return self._asarray(results)
 
+    def batch_execute(self, circuits):
```

Comment:
Do we definitely want to call this `batch_execute()` and not something like `execute_circuits()`/`execute_tapes()`?

Reply:
I like the fact that it is agnostic of what is its input - always a good choice. And "batch-job" is actually a technical term, see for example [here](https://www.bmc.com/blogs/batch-jobs/). It describes quite well what we want - we submit a whole lot of instructions and the backend figures out what to do with them.